Scripts/RemediationScripts/Remediate-EnableTransparentDataEncryption.ps1

```diff
@@ -0,0 +1,653 @@
+
+        # Get all SQL Servers in a Subscription      
+        $sqlServersResources = Get-AzSqlServer
+        $totalSqlServers = ($sqlServersResources | Measure-Object).Count
+
+        if ($totalSqlServers -eq 0)
+        {
+            Write-Host "No SQL Server found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    
+        Write-Host "Found $($totalSqlServers) SQL Server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "Fetching SQL Server configurations..."
+
+        
+
+        $sqlServersResources | ForEach-Object{
+            $databaseList = Get-AzSqlDatabase -ServerName $_.ServerName -ResourceGroupName $_.ResourceGroupName
+            $databaseList | ForEach-Object{
+                if ($_.DatabaseName -ne 'master')
+                {
+                    $tdeDetails = Get-AzSqlDatabaseTransparentDataEncryption -ServerName $_.ServerName -DatabaseName $_.DatabaseName -ResourceGroupName $_.ResourceGroupName
+                    if ($tdeDetails.State -eq "Enabled")
+                    {
+                        $sqlDatabaseWithTDEEnabled += $tdeDetails
+                    }
+                    else
+                    {
+                        $sqlDatabaseWithTDEDisabled += $tdeDetails
+                    }
+                }                
+            }
+        }
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all SQL Servers from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        $sqlDatabaseWithTDEDisabled = Import-Csv -LiteralPath $FilePath
+    }
+
+    $totalSqlDatabaseWithTDEDisabled = ($sqlDatabaseWithTDEDisabled | Measure-Object).Count
+
+    if ($totalSqlDatabaseWithTDEDisabled -eq 0)
+    {
+        Write-Host "No SQL Serrver database found with Transparent Data Encryption (TDE) disabled. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalSqlDatabaseWithTDEDisabled) SQL Server database(s) with Transparent Data Encryption (TDE) disabled." -ForegroundColor $([Constants]::MessageType.Update)
+
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableTDEForSQLServers"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up SQL Server details to $($backupFolderPath)"
+    
+    # Backing up SQL Server details.
+    $backupFile = "$($backupFolderPath)\SqlServersWithTDEDisabled.csv"
```

Comment:
If we are going to log individual SQL Database level records, the file name can be renamed to reflect this.